container-core/src/main/java/com/yahoo/container/handler/LogHandler.java

```diff
@@ -60,26 +61,57 @@ public void render(OutputStream output, ContentChannel networkChannel, Completio
     }
 
 
-    private static class BlockingFlushContentChannelOutputStream extends ContentChannelOutputStream {
-
+    private static class MaxPendingContentChannelOutputStream extends ContentChannelOutputStream {
         private final ContentChannel channel;
+        private final long maxPending;
+        private AtomicLong sent = new AtomicLong(0);
+        private AtomicLong acked = new AtomicLong(0);
 
-        public BlockingFlushContentChannelOutputStream(ContentChannel endpoint) {
+        public MaxPendingContentChannelOutputStream(ContentChannel endpoint, long maxPending) {
             super(endpoint);
             this.channel = endpoint;
+            this.maxPending = maxPending;
+        }
+
+        private long pendingBytes() {
+            return sent.get() - acked.get();
```

Comment:
Consider replacing with a single atomic that is decremented for `completed()`/`failed()`

Reply:
I have concluded that most of the time it is better to have separate counters both counting up.
1 - There will be less contention.
2 - You preserve more information both for debugging and if you want monitoring.